### DISCUSSION TITLE
Is it possible to reuse workflow yaml to setup similar workflows?

### DISCUSSION THREAD

**Discussion by smac89** [community/community#25504] [Actions]
<p>Basically, is there any support for using yaml files as templates for other workflow yaml files?</p>
<p>Gitlab has a similar feature using the <code>include</code> keyword to include workflow templates; and in addition the <code>extends</code> keyword can be used to share small bits of yaml within the same yaml file</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Thanks <a class="mention" href="/smac89">@smac89</a> , workflow template or keyword like ‘extends’ is not supported in Github currently, please create individual workflow file for your configuration.</p>

**Comment by johnc-ftl** [comment]
<p>This may not have been available in Dec '19, but a “Composite Run Steps Action” (just flows off the tongue, doesn’t it?) is probably your best option for re-using YAML logic in Github actions:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-composite-run-steps-action" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-composite-run-steps-action" target="_blank" rel="noopener nofollow ugc">Creating a composite run steps action - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
